src/UXClient/Components/Marker/Marker.ts

```diff
@@ -0,0 +1,274 @@
+                            if (d3.select(d3.event.sourceEvent.target).classed('tsi-closeButton')) {
+                                return;
+                            }
+                            let marker = d3.select(<any>d3.select(this).node().parentNode);
+                            let startPosition = self.x(new Date(self.timestampMillis));
+                            let newPosition = startPosition + d3.event.x;
+
+                            self.timestampMillis = Utils.findClosestTime(self.x.invert(newPosition).valueOf(), self.chartComponentData.timeMap);
+                            self.setPositionsAndLabels(self.timestampMillis);
+                        })
+                        .on('end', function (d) {
+                            if (!d3.select(d3.event.sourceEvent.target).classed('tsi-closeButton')) {
+                                self.onChange(false, false);
+                            }
+                        })
+                    );
+            });
+        marker.exit().remove();
+    }
+
+    private getValueOfVisible (d: any) {
+        return Utils.getValueOfVisible(d, this.chartComponentData.getVisibleMeasure(d.aggregateKey, d.splitBy))
+    }
+
+    private getTooltipKey (d: any) {
+        return d.aggregateKey + '_' + d.splitBy;
+    }
+
+    private setValueLabels (closestTime) {
+        let values = this.chartComponentData.timeMap[closestTime] != undefined ? this.chartComponentData.timeMap[closestTime] : [];
+
+        values = values.filter((d) => {
+            return (this.getValueOfVisible(d) !== null) && this.chartComponentData.getDataType(d.aggregateKey) === DataTypes.Numeric; 
+        });
+        let self = this;
+
+        let valueLabels = this.markerContainer.selectAll(".tsi-markerValue").data(values, (d) => {
+            return d.aggregateKey + "_" + d.splitBy;
+        });
+
+        valueLabels.enter()
+            .append("div")
+            .classed("tsi-markerValue", true)
+            .merge(valueLabels)
+            .style('top', (d) => this.calcTopOfValueLabel(d) + 'px')
+            .each(function (d: any) {
+                let tooltipKey = self.getTooltipKey(d);
+                let tooltip;
+                
+                if (self.tooltipMap[tooltipKey]) {
+                    tooltip = self.tooltipMap[tooltipKey];
+                } else {
+                    tooltip = new Tooltip(d3.select(this));
+                    self.tooltipMap[tooltipKey] = tooltip;
+                }
+                tooltip.render(self.chartOptions.theme);
+                let tooltipHeight = MARKERVALUENUMERICHEIGHT;
+                tooltip.draw(d, self.chartComponentData, 0, tooltipHeight/2, {right:0, left:0, top:0, bottom:0}, (tooltipTextElement) => {
+
+                    tooltipTextElement.text(Utils.formatYAxisNumber(self.getValueOfVisible(d)))
+                        .classed('tsi-markerValueTooltipInner', true)
+                        .style('height', tooltipHeight + 'px')
```

Comment:
curious if stuff like this cant entirely be handled by CSS using classnames determined from properties that are injected in tooltip.render or tooltip.draw?  conceptually like tooltip.render(self.chartOptions.theme, {isForMarker: true})